rfc/Errors-Extension-Specification.md

```diff
@@ -0,0 +1,123 @@
+# Subgraph Errors Extension Specification
+
+## Problem
+
+For security reasons, we're currently not exposing Subgraph Errors to the client.
+Subgraphs might leak sensitive information in the error object, e.g. through the `extensions` field.
+
+What we've learned from our users is that they would like to leverage the `extensions` field to provide more context to the client,
+e.g. by adding a `code` field to the error object.
+
+A response object might look like this:
+
+```json
+{
+  "errors": [
+    {
+      "message": "Cannot query field 'foo' on type 'Bar'",
+      "path": [
+        "foo"
+      ],
+      "extensions": {
+        "code": "GRAPHQL_VALIDATION_FAILED"
+      }
+    }
+  ]
+}
+```
+
+With this information, the client can now handle the error more gracefully.
+
+However, this leads to the problem that Subgraph owners might send "anything" in the `extensions` field,
+which might leak sensitive information.
+
+Another problem is that users would like to standardize the error codes across different Subgraphs,
+e.g. `GRAPHQL_VALIDATION_FAILED` should always mean the same thing,
+and there should be agreement across all Subgraph owners on what codes to use.
+
+Additional fields in the `extensions` object could also be allowed, e.g. `message` or `severity`,
+but it's important that all Supergraph contributors agree on what fields and values are allowed.
+
+## Solution
+
+We propose to introduce the ability to define a schema for the `extensions` field in the Supergraph.
+As the response encoding is JSON, we can leverage JSON Schema to define the schema for the error object.
+
+The schema can be attached at the Supergraph level using the `wgc` cli tool using the following command:
+
+```bash
+npx wgc federated-graph create production -r http://router.example.com/graphql --label-matcher team=A department=backend --errors-extensions-schema ./errors-extensions-schema.json
+```
+
+Example of updating the schema for an existing Supergraph:
+
+```bash
+npx wgc federated-graph update production --errors-extensions-schema ./errors-extensions-schema.json
+```
+
+The schema file should look like this:
+
+```json
+{
```

Comment:
How would conflicting definitions between subgraphs be handled in this case? I.e different values for `required`?

Reply:
Nevermind I misread `subgraph` for `subgraph` on L52.